Add product lifecycle page and regions page

mirnawong1 opened the pull request:
## Description & motivation
Added a new page for product lifecycles and regions
Closes #2135 and #2134 


## To-do before merge

- [x] Update any content from google docs
- [x] Link to new region page in the following docs:
* https://docs.getdbt.com/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database#ip-restrictions
* https://docs.getdbt.com/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-enterprise-snowflake-oauth#server-error-500


## Prerelease docs
If this change is related to functionality in a prerelease version of dbt (delete if not applicable):
- [ ] I've added versioning components, as described in ["Versioning Docs"](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md)
- [ ] I've added a note to the prerelease version's [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

## Checklist
If you added new pages (delete if not applicable):
- [x ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`
- [ ] [Run link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

mirnawong1 committed:
Create product-lifecycles.md

mirnawong1 committed:
Update sidebars.js

added product lifecycle new page to sidebar

mirnawong1 committed:
Create regions.md

added new ip address page

mirnawong1 committed:
Update sidebars.js

mirnawong1 committed:
Update sidebars.js

JKarlavige commented on website/sidebars.js:
```diff
@@ -276,6 +277,7 @@ const sidebarSettings = {
       items: [
         "docs/dbt-versions/core",
         "docs/dbt-versions/upgrade-core-in-cloud",
+        "docs/dbt-versions/product-lifecycles"
```
comma needs to be after this line: `"docs/dbt-versions/product-lifecycles",`

mirnawong1 committed:
Update sidebars.js

mirnawong1 committed:
Update product-lifecycles.md

mirnawong1 committed:
Update regions.md

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Merge branch 'next' into mwong-lifecyle-region-new-pages

mirnawong1 committed:
Update regions.md

mirnawong1 committed:
Update product-lifecycles.md

mirnawong1 committed:
Update set-up-snowflake-oauth.md

mirnawong1 committed:
Update regions.md

mirnawong1 commented:
hi @azzam34, made some tweaks to the language so it has some active voice changes (rather than passive). i can tell some of the product lifecycle stages are in passive voice for a reason so i changed some content to active if it didn't have anything to do with the lifecycle state.

i also added plan availability to the regions content. the reason why is - if we're linking to the regions page on other pages like the [IP restrictions section](https://docs.getdbt.com/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database#ip-restrictions, then i think we also need to treat the page as a 'regions and IP' page, which is an important reference to dev, team and enterprise users. (there are a lot of users that reach out to support asking for IP addresses)

please let me know what you think and provide any feedback, thank you!

azzam34 commented on website/docs/docs/dbt-versions/product-lifecycles.md:
```diff
@@ -0,0 +1,47 @@
+
+- **Beta:** Beta features may be made available for the purpose of customer testing and evaluation. These may not be feature-complete or fully stable. There may still be some planned additions and modifications to product behaviors while in Beta. Breaking changes may occur – although we will do our best to communicate them in advance, we may not always be able to do so. Beta features may not be f ...
+
+- **Preview (Private or Public):**  Preview features are stable and can be considered for production deployments. There may still be some planned additions and modifications to product behaviors before moving to General Availability. We may also introduce new functionality to Preview features that are not backward compatible. Preview features include documentation, technical support, and include  ...
```
Should say: "We may also introduce new functionality to Preview features that **is** not backward compatible"

mirnawong1 replied:
ah ok, so i changed it to **are** because "features" is plural but i think both are fine so will add your sugg

azzam34 commented on website/docs/docs/dbt-versions/product-lifecycles.md:
```diff
@@ -0,0 +1,47 @@
+
+- **Preview (Private or Public):**  Preview features are stable and can be considered for production deployments. There may still be some planned additions and modifications to product behaviors before moving to General Availability. We may also introduce new functionality to Preview features that are not backward compatible. Preview features include documentation, technical support, and include  ...
+
+- **Generally Available(GA):** Generally Available features are stable and can be considered for production deployments. GA features include documentation, technical support, and service level agreements (SLAs) apply.
```
Not a huge deal by any means, but I did prefer the phrasing in the original doc:

> Features are fully documented, technical support is provided, and service level agreements (SLAs) apply.

mirnawong1 replied:
no worries @azzam34 , just switched it to active voice because it's more beneficial for translation (in the future) and non English speakers. @runleonarun is also reviewing this too - Leona if you can also check this too please? 🙏

runleonarun replied:
Here's my suggestion!
```suggestion
- **Generally Available(GA):** Generally Available features provide stable features that can be considered for production deployments. Service level agreements (SLAs) apply to GA features, and these features include documentation and technical support.
```
I have a question, @azzam34 and @mirnawong1  should we be promising support for new features? I thought support depended on your plan? Also I would refrain from saying "fully documented" just in case that means more than what we can provide with such a small team.

mirnawong1 replied:
I'll defer to @barberscott for the support question. my understanding is that for GA releases, the user can reach out to support, which may or may not lead to bug reports from the support team after they investigate.

azzam34 commented on website/docs/docs/dbt-versions/product-lifecycles.md:
```diff
@@ -0,0 +1,47 @@
+
+### dbt Core
+
+- **Unreleased:** We include this functionality in the next minor version prerelease. However, we make no commitments about its behavior or implementation. As maintainers, we reserve the right to change any part of it, or remove it entirely (with an accompanying explanation.)
```
We *will

mirnawong1 replied:
```suggestion
- **Unreleased:** We will include this functionality in the next minor version prerelease. However, we make no commitments about its behavior or implementation. As maintainers, we reserve the right to change any part of it, or remove it entirely (with an accompanying explanation.)
```

mirnawong1 commented on website/docs/docs/dbt-versions/product-lifecycles.md:
```diff
@@ -0,0 +1,47 @@
+
+- **Beta:** Beta features may be made available for the purpose of customer testing and evaluation. These may not be feature-complete or fully stable. There may still be some planned additions and modifications to product behaviors while in Beta. Breaking changes may occur – although we will do our best to communicate them in advance, we may not always be able to do so. Beta features may not be f ...
+
+- **Preview (Private or Public):**  Preview features are stable and can be considered for production deployments. There may still be some planned additions and modifications to product behaviors before moving to General Availability. We may also introduce new functionality to Preview features that are not backward compatible. Preview features include documentation, technical support, and include  ...
```
```suggestion
- **Preview (Private or Public):**  Preview features are stable and can be considered for production deployments. There may still be some planned additions and modifications to product behaviors before moving to General Availability. We may also introduce new functionality to Preview features that is not backward compatible. Preview features include documentation, technical support, and include service level objectives (SLOs). Features in Preview are generally provided at no extra cost, although they may become paid features in their Generally Available state.
```

mirnawong1 replied:
```suggestion
- **Preview (Private or Public):**  Preview features are stable and can be considered for production deployments. There may still be some planned additions and modifications to product behaviors before moving to General Availability. We may also introduce new functionality to Preview features that is not backward compatible. Preview features include documentation, technical support, and include service level objectives (SLOs). Features in Preview are generally provided at no extra cost, although they may become paid features in their Generally Available state.
```

mirnawong1 committed:
Update website/docs/docs/dbt-versions/product-lifecycles.md

mirnawong1 committed:
Update website/docs/docs/dbt-versions/product-lifecycles.md

runleonarun commented on website/docs/docs/dbt-versions/product-lifecycles.md:
```diff
@@ -0,0 +1,47 @@
+- **Removed:** Removed features no longer have any level of product functionality or platform support.
+
+### dbt Core
+
```
We might want to point to the versioning page here. 
```suggestion

We release dbt Core in the following lifecycle states. Core releases follow semantic layer versioning, which you  can read more about in [About Core versions](/docs/dbt-versions/core).
```

runleonarun commented on website/docs/docs/deploy/regions.md:
```diff
@@ -0,0 +1,19 @@
+[dbt Cloud Enterprise](https://www.getdbt.com/pricing/) plans can choose to have their account hosted in any of the below regions. Organizations **must** choose a single region per dbt Cloud account. If you need to run dbt Cloud in multiple regions, we recommend using multiple dbt Cloud accounts. 
+
+
+| Region | Location | Access URL | IP addresses | Developer plan | Team plan | Enterprise plan |
```
Love this table @mirnawong1 !

runleonarun reviewed (approved):
@mirnawong1 and @azzam34 this is so great! I just had some minor suggestions so feel free to merge into next once these are addressed!

mirnawong1 committed:
Update website/docs/docs/dbt-versions/product-lifecycles.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'next' into mwong-lifecyle-region-new-pages

mirnawong1 committed:
Update website/docs/docs/dbt-versions/product-lifecycles.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>
